pkg/scheduler/scheduler.go

```diff
@@ -684,3 +688,227 @@ func newPodInformer(cs clientset.Interface, resyncPeriod time.Duration) cache.Sh
 	}
 	return coreinformers.NewFilteredPodInformer(cs, metav1.NamespaceAll, resyncPeriod, nil, tweakListOptions)
 }
+
+// numFeasibleNodesToFind returns the number of feasible nodes that once found, the scheduler stops
+// its search for more feasible nodes.
+func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes int32) {
```

Comment:
The goal is not to simply move the functions from generic_scheduler.go to scheduler.go. We want to eliminate `genericScheduler` entirely. Could you check the discussion of #101135 to understand the intents first?

Reply:
Thanks for the review!
I understand that the `genericScheduler` and `SchedulerAlgorithm` should be remove finally, and I prepared to merge `genericScheduler` to `Scheduler` and change the functions in later patch. 
So I should move functions and fit it filename in the same patch,  is that right?

Reply:
The two actions should be in separate PRs (or at least commits)

I think it's more useful to start removing the `genericScheduler` without moving the functions between files. This would help us know if we missed a crucial detail.

Reply:
> The two actions should be in separate PRs (or at least commits)
> 
> I think it's more useful to start removing the `genericScheduler` without moving the functions between files. This would help us know if we missed a crucial detail.

Thanks for your advice. That sounds really reasonable. 

I will remove the `genericScheduler` in another PR.